# CustomerAddress Sheet

In [1]:
import numpy as np
import pandas as pd

In [6]:
custAd = pd.read_excel("KPMG_VI_New_raw_data_update_final.xlsx", sheet_name= "CustomerAddress",skiprows=1)
custAd.head(10)

customer_id                  address  postcode            state    country  \
0            1       060 Morning Avenue      2016  New South Wales  Australia   
1            2      6 Meadow Vale Court      2153  New South Wales  Australia   
2            4       0 Holy Cross Court      4211              QLD  Australia   
3            5      17979 Del Mar Point      2448  New South Wales  Australia   
4            6         9 Oakridge Court      3216              VIC  Australia   
5            7         4 Delaware Trail      2210  New South Wales  Australia   
6            8      49 Londonderry Lane      2650  New South Wales  Australia   
7            9          97736 7th Trail      2023  New South Wales  Australia   
8           11     93405 Ludington Park      3044              VIC  Australia   
9           12  44339 Golden Leaf Alley      4557              QLD  Australia   

   property_valuation  
0                  10  
1                  10  
2                   9  
3                   4  
4                   9  
5                   9  
6                   4  
7                  12  
8                   8  
9                   4

In [33]:
custAd.shape

(3999, 6)

### State Column

In [11]:
custAd[['state']].head(10)

state
0  New South Wales
1  New South Wales
2              QLD
3  New South Wales
4              VIC
5  New South Wales
6  New South Wales
7  New South Wales
8              VIC
9              QLD

In [13]:
custAd['state'].unique()

array(['New South Wales', 'QLD', 'VIC', 'NSW', 'Victoria'], dtype=object)

We can see here data inconsistency i.e New South Wales and NSW. Both full name and codename used in the dataset.

### Property Valuation Column

In [17]:
custAd[['property_valuation']].head()

property_valuation
0                  10
1                  10
2                   9
3                   4
4                   9

No meaningful information derived from this field means that no currency sign/value is mentioned i.e we can not say that is 10 is either 10 million dollars or euro or if 10 billion.

### Join with Customer Demographic Table

In [36]:
custDemo = pd.read_excel("KPMG_VI_New_raw_data_update_final.xlsx", sheet_name= "CustomerDemographic",skiprows=1)
custDemo.head()

customer_id      first_name  last_name  gender  \
0            1         Laraine  Medendorp       F   
1            2             Eli    Bockman    Male   
2            3           Arlin     Dearle    Male   
3            4          Talbot        NaN    Male   
4            5  Sheila-kathryn     Calton  Female   

   past_3_years_bike_related_purchases        DOB               job_title  \
0                                   93 1953-10-12     Executive Secretary   
1                                   81 1980-12-16  Administrative Officer   
2                                   61 1954-01-20      Recruiting Manager   
3                                   33 1961-10-03                     NaN   
4                                   56 1977-05-13           Senior Editor   

  job_industry_category     wealth_segment deceased_indicator  \
0                Health      Mass Customer                  N   
1    Financial Services      Mass Customer                  N   
2              Property      Mass Customer                  N   
3                    IT      Mass Customer                  N   
4                   NaN  Affluent Customer                  N   

                                             default owns_car  tenure  
0                                                 "'      Yes    11.0  
1                       <script>alert('hi')</script>      Yes    16.0  
2                                2018-02-01 00:00:00      Yes    15.0  
3  () { _; } >_[$($())] { touch /tmp/blns.shellsh...       No     7.0  
4                                                NIL      Yes     8.0

In [35]:
custDemo[~custDemo['customer_id'].isin(custAd['customer_id'])]

customer_id first_name last_name  gender  \
2             3      Arlin    Dearle    Male   
9            10   Fiorenze   Birdall  Female   
21           22    Deeanne  Durtnell  Female   
22           23       Olav     Polak    Male   

    past_3_years_bike_related_purchases        DOB                job_title  \
2                                    61 1954-01-20       Recruiting Manager   
9                                    49 1988-10-11  Senior Quality Engineer   
21                                   79 1962-12-10                      NaN   
22                                   43 1995-02-10                      NaN   

   job_industry_category  wealth_segment deceased_indicator  \
2               Property   Mass Customer                  N   
9     Financial Services   Mass Customer                  N   
21                    IT   Mass Customer                  N   
22                   NaN  High Net Worth                  N   

                default owns_car  tenure  
2   2018-02-01 00:00:00      Yes    15.0  
9                  ð©ð½      Yes    20.0  
21               ï¼ï¼ï¼       No    11.0  
22  2018-02-01 00:00:00      Yes     1.0

We can see here that customer_id number **3, 10, 22, 23** records from **CustomerDemographic** table/sheet are not in **CustomerAddress** table/sheet.

In [37]:
custAd[~custAd['customer_id'].isin(custDemo['customer_id'])]

customer_id                 address  postcode state    country  \
3996         4001  87 Crescent Oaks Alley      2756   NSW  Australia   
3997         4002        8194 Lien Street      4032   QLD  Australia   
3998         4003         320 Acker Drive      2251   NSW  Australia   

      property_valuation  
3996                  10  
3997                   7  
3998                   7

Also customer_id number **4001, 4002, 4003** records from **CustomerAddress** table/sheet are not in** CustomerDemographic** table/sheet.